huxley/api/permissions.py

```diff
@@ -91,6 +95,35 @@ def has_permission(self, request, view):
         return False
 
 
+class DelegateListPermission(permissions.BasePermission):
+
+    def has_permission(self, request, view):
+        user = request.user
+        if user.is_superuser:
+            return True
+
+        method = request.method
+        if method in permissions.SAFE_METHODS:
+            return user_is_advisor(request, view)
+
+        if method in ('POST', 'PUT', 'PATCH'):
+            school_id = user.is_authenticated() and user.school_id
+            if not school_id:
+                False
+
+            if method == 'POST':
+                return int(request.data['school']) == school_id
+            else:
+                data = request.data
+                if isinstance(data, QueryDict):
```

Comment:
Why do we need this hack again? I remember you telling me, but I forgot the details.